inference-engine/tests/functional/plugin/conformance/subgraphs_dumper/include/matchers/base_matcher.hpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "pugixml.hpp"
+#include "ngraph/node.hpp"
+
+namespace SubgraphsDumper {
+
+class MatchersManager;
+
+class iMatcherConfig {
+public:
+    using Ptr = std::shared_ptr<iMatcherConfig>;
+
+    // Empty vectors stands for any of possible values
+    std::vector<std::string> ignored_attributes;
+    std::vector<size_t> ignored_ports;
+    bool is_fallback_config;
+
+    virtual bool op_in_config(const std::shared_ptr<ngraph::Node> &node) = 0;
+};
+
+template<typename ...OPTypes>
+struct MatcherConfig : public iMatcherConfig {
+public:
+    MatcherConfig() : iMatcherConfig() {
+        if (sizeof...(OPTypes) == 0) {
+            is_fallback_config = true;
+        }
+    }
+
+    MatcherConfig(const std::vector<std::string> &_ignored_attributes,
+                  const std::vector<size_t> &_ignored_ports) : iMatcherConfig() {
+        ignored_attributes = _ignored_attributes;
+        ignored_ports = _ignored_ports;
+
+        if (sizeof...(OPTypes) == 0) {
+            is_fallback_config = true;
+        }
```

Comment:
`iMatcherConfig` is not templated class. I can't make it templated because  `default_config` attribute  of `MatchersManager` should store just a pointers to the abstract class without template arguments specification. 
Without having template such kind of check for setting `is_fallback_config` can't be performed in `iMatcherConfig`.

That's how I see it, unless I'm missing something.

Reply:
I thins to have something like this if is this possible:
```
    MatcherConfig(std::vector<std::string> ignored_attributes,
                  std::vector<size_t> ignored_ports) 
        : iMatcherConfig(std::move(ignored_attributes), 
                         std::move(ignored_ports),
                         sizeof...(OPTypes) == 0) {
    }
```

Reply:
It's a good option, didn't think about it. Thanks, I will do it so